cni.go

```diff
@@ -137,9 +137,19 @@ func (c *libcni) Load(opts ...Opt) error {
 func (c *libcni) Status() error {
 	c.RLock()
 	defer c.RUnlock()
-	if len(c.networks) < c.networkCount {
-		return ErrCNINotInitialized
+	if err := c.ready(); err != nil {
+		return err
+	}
+
+	// STATUS is only called for CNI Version 1.1.0 or greater. It is ignored for previous versions.
+	for _, v := range c.networks {
+		err := c.cniConfig.GetStatusNetworkList(context.Background(), v.config)
+
+		if err != nil {
+			return err
```

Comment:
looking at the crio impl they chose to only check the default/primary network:
`
func (plugin *cniNetworkPlugin) StatusWithContext(ctx context.Context) error {
	defaultNet := plugin.getDefaultNetwork()
`
I think that is fair

Reply:
and we can log the others

Reply:
The caller in this case containerd will need to log this. I can do that

Reply:
yeah well if only the caller can log then we have no choice but to only check primary.. we'll need another api to gather status for all

Reply:
iow as written by the crio team the status is only for  the default network not all of the networks..

Reply:
In the current form, say WithDefaultConf is used we only get the default network here since loopback would be skipped. The max number of networks is 2 leaving the primary to give status. 

If WithAllConf is used we would go through all networks and ensure all status is not nil. containers uses the WithDefaultConf approach.

Reply:
looks good I got twisted up with the map of cni instances vs map of networks in each instance..